### PR TITLE
Fix broken links and add linkcheck back to CI

### PR DESCRIPTION
**Pull request by wihobbs**
Problem: there are some broken links in our docs.

Fix them.

### CODE REVIEW

File: jobs/batch.rst [hunk 1]
```diff
@@ -6,9 +6,9 @@ Batch Jobs
 
 In Flux, a batch job is a script submitted along
 with a request for resources. When resources become
-available, the script is run as the
-`initial program <https://flux-framework.readthedocs.io/projects/flux-rfc/en/latest/spec_8.html#initial-program-program-1>`_ of a new single-user instance of Flux,
```
Comment (jameshcorbett) [inline jobs/batch.rst:10]: https://flux-framework.readthedocs.io/projects/flux-core/en/latest/guide/glossary.html#term-initial-program

https://flux-framework.readthedocs.io/projects/flux-core/en/latest/guide/glossary.html#term-instance-owner

**Review by jameshcorbett — APPROVED** [review]
Looks reasonable, there are two links that just need to be changed I think rather than removed.

**Comment by jameshcorbett** [comment]
Oh sorry didn't see this was a draft :|

**Comment by wihobbs** [comment]
eh maybe it doesn't have to be. I'd like to get back to automating linkcheck with CI but maybe we can accomplish that in a separate pr? Thoughts, @jameshcorbett?

**Comment by jameshcorbett** [comment]
It depends on how much work there is to be done. If you think you could do it without a ton more effort that would be nice, because the proper way to review this PR as it is now would be to manually verify that all the links work which is tedious

**Review by jameshcorbett — APPROVED** [review]
💯

**Comment by wihobbs** [comment]
Thanks @jameshcorbett! I'll set MWP.

**Pull request merged**